students/hyska_monika/lesson_12_Debugging/csv_functions.py

```diff
@@ -0,0 +1,40 @@
+import csv
+import os
+import logging
+
+
+# save data to new line or replace row when data from fist column is the same
+def save_data_to_csv(filename, header, data):
+    logging.info('filename: %s, header: %s, data: %s'
+                 % (filename, header, data))
+    if not (os.path.isfile(filename)):
+        with open(filename, 'w', newline='') as emty_csv:
+            writer = csv.writer(emty_csv)
+            logging.info('empty csv file: %s crated' % filename)
+            writer.writerow(header)
+            logging.info('header - columns name: (%s) are saved' % header)
+    with open(filename) as inf:
+        reader = csv.reader(inf.readlines())
+    with open(filename, 'w', newline='') as outf:
+        writer = csv.writer(outf)
+        repalced = 0
+        for line in reader:
+            logging.info('Line in file: %s' % line)
+            if not line:
+                logging.error('IndexError: list index out of range. '
```

Comment:
It might be confusing to use exception names, there is `logging.exception` to be used in except section for that. It will save traceback also.